### PR TITLE
Make Thicket Construction Print Statements Optional

### PR DESCRIPTION
**Pull request by michaelmckinsey1**
Referring to these:
```
Union Graph == thicket[1].graph
Union Graph == thicket[1].graph
Union Graph == thicket[1].graph
```
With large datasets, this becomes an annoyance, see https://lc.llnl.gov/gitlab/pave/thicket/-/issues/23

### CODE REVIEW

File: thicket/thicket.py [hunk 1]
```diff
@@ -683,24 +683,41 @@ def unify_pair(self, other):
 
         return union_graph
 
-    def unify_pairwise(th_list):
+    @staticmethod
```
Comment (slabasan) [inline thicket/thicket.py:686]: Do these functions need to be static? If not, we should make another PR to remove this specification.

File: thicket/thicket.py [hunk 2]
```diff
@@ -683,24 +683,41 @@ def unify_pair(self, other):
...
+    def unify_pairwise(th_list, debug=False):
```
Comment (slabasan) [inline thicket/thicket.py:687]: If we are going to have more debug statements throughout thicket, we might think about having a global enable/disable debug mode that we can specify from an environment variable or config file.

**Review by slabasan — COMMENTED** [review]
LGTM, just 1 question about staticmethods.

File: thicket/thicket.py (hunk 1 above)
Comment (michaelmckinsey1) [reply thicket/thicket.py:686]: My current strategy is to use this decorator whenever we are not using `self` inside a class function. Otherwise, you will run into [errors](https://stackoverflow.com/a/43587154) when trying to call the function outside of the class.

I suppose we could remove them from any methods in which we don't want any users to run. But then would we want to prepend nothing, `_`, or `__` before these functions?

File: thicket/thicket.py (hunk 2 above)
Comment (michaelmckinsey1) [reply thicket/thicket.py:687]: If we want to use [dotenv](https://pypi.org/project/python-dotenv/) I can add that here. I don't think any other functions are using a debug variable at the moment.

Comment (slabasan) [reply thicket/thicket.py:687]: We don't need to implement it now, just something for us to keep in mind if we find we want to have more debugging capabilities.

File: thicket/thicket.py (hunk 1 above)
Comment (slabasan) [reply thicket/thicket.py:686]: Got it, makes sense.
